src/internal/clusterstate/v2.8.0/pfsdb.go

```diff
@@ -212,6 +212,102 @@ func migrateRepos(ctx context.Context, tx *pachsql.Tx) error {
 	return nil
 }
 
+// alterCommitsTable adds useful new columns to pfs.commits table.
+// Note that this is not the end all be all. We will need to make more changes after data has been migrated.
+// TODO
+//   - rename int_id to id? This will requires changing all references as well.
+//   - make repo_id not null
+//   - make origin not null
+//   - make updated_at not null and default to current timestamp
+//   - branch_id_str is a metadata reference to the branches table. Today this points to collections, but tomorrow it should
+//     point to pfs.branches. Once the PFS master watches branches, we will no longer need this column.
+func alterCommitsTable(ctx context.Context, tx *pachsql.Tx) error {
+	query := `
+	CREATE TYPE pfs.commit_origin AS ENUM ('ORIGIN_KIND_UNKNOWN', 'USER', 'AUTO', 'FSCK');
+
+	ALTER TABLE pfs.commits
+		ADD COLUMN repo_id bigint REFERENCES pfs.repos(id),
+		ADD COLUMN origin pfs.commit_origin,
+		ADD COLUMN description text NOT NULL DEFAULT '',
+		ADD COLUMN start_time timestamptz NOT NULL DEFAULT CURRENT_TIMESTAMP,
+		ADD COLUMN finishing_time timestamptz,
+		ADD COLUMN finished_time timestamptz,
+		ADD COLUMN compacting_time bigint,
```

Comment:
I think there's a slack thread where we talk about adding units to this column?

Reply:
does that mean we'd want to represent this as a string? or is there a time code that includes the units?

Reply:
I think Brendon meant does is the bigint in units of `ns, ms, s`?

Reply:
Yeah or `compacting_time_s`